src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,44 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  bite(anim) {
+    if (anim instanceof Herbivore && anim.hidden === false) {
+      anim.health -= 50;
+    }
+
+    if (anim.health === 0) {
+      const indexAnim = Animal.alive.indexOf(anim);
+
+      Animal.alive.splice(indexAnim, 1);
```

Comment:
You also could write it without second index
```suggestion
      Animal.alive.splice(indexAnim);
```